replica/database.hh

```diff
@@ -1514,6 +1514,7 @@ public:
     const gms::feature_service& features() const { return _feat; }
     future<> apply_in_memory(const frozen_mutation& m, schema_ptr m_schema, db::rp_handle&&, db::timeout_clock::time_point timeout);
     future<> apply_in_memory(const mutation& m, column_family& cf, db::rp_handle&&, db::timeout_clock::time_point timeout);
+    future<> unfreeze_and_apply_in_memory(const frozen_mutation& m, schema_ptr m_schema, column_family& cf, db::rp_handle h, db::timeout_clock::time_point timeout);
```

Comment:
If there are no other callers planned for this method, let's just inline it into its single caller. The database API is crowded enough already.

Reply:
@denesb the problem is that the caller uses continuations and we don't to pessimize it and makexthe whole function a coroutine.
If only this slow path would be inlined we'll either need to use do_with or have a coroutine lambda function which I'm not sure is simpler than this method (which is a coroutine)

Reply:
I don't think we should go as far as extract arbitrary pieces of code into methods, just to make them coroutines. This is two lines of code, in-lining as continuations is not that bad.

Reply:
Ixm afraid that the continuations may add an allocation for do_with on top of the additional tasks/allocation if unfreeze_gently returns an unavailable future (as expected), therefore a coroutune should be slightly more efficient.